zulip/integrations/slack/slackdata2zulipdata.py

```diff
@@ -339,21 +339,38 @@ def main(slack_zip_file: str) -> None:
     zerver_message = []
     zerver_usermessage = []
 
+    # check if a user has been mentioned in a message
+    def check_user_mention(text):
+        mentions = re.findall(r'(@(?:\*\*([^\*]+)\*\*|(\w+)))', text)
+        mentioned_users_id = []
+        for mention in mentions:
+            for userprofile in zerver_userprofile:
+                if mention[1] == userprofile['full_name']:
+                        mentioned_users_id.append(userprofile['id'])
+        return mentioned_users_id
+
     for channel in added_channels.keys():
         zerver_message.append(channelmessage2zerver_message(slack_dir, channel,
                               added_users, added_channels))
+
     # construct the usermessage object and append it to zerver_usermessage
     usermessage_id = 1
-    for zulip_message in zerver_message:
-        for subscription in zerver_subscription:
-            if subscription['recipient'] == zulip_message['recipient']:
-                usermessage = dict(
-                    user_profile=subscription['user_profile'],
-                    id=usermessage_id,
-                    flags_mask=1,  # defaulting to 'read'
-                    message=zulip_message['id'])
-                usermessage_id += 1
-                zerver_usermessage.append(usermessage)
+    for message_index in range(len(zerver_message)):
```

Comment:
Why the extra for loop? I couldn't see `message_index` being use elsewhere.

Reply:
It looks to me as:
```python
for i in a:
    for j in a:
        ...
```

Reply:
So zerver_message is a list of list of dictionary. So we need two for loops to traverse for every single dictionary.

Reply:
That was the error I made initially. I assumed it to be a list of dicts. But for every channel, a list of dicts is being appended in the zerver_message list.